### PR TITLE
Add MongoMetricsConnectionPoolTagsProvider

### LINKED ISSUE #2355: Support for Connection pool tags provider
This is especially important in applications which have multiple mongo connection pools, for example.

Thanks in advance, and great job/project!

### PR DESCRIPTION
**Pull request by gustavomonarin**
This is extremely useful for example to differentiate the connection pool metrics in an application with multiple different connections to different databases which could be in the same server.

resolves #2355

### CODE REVIEW

**Review by bono007 — APPROVED** [review]
LGTM @gustavomonarin - thanks for the update - I do like the consistency between this and #2360.

File: micrometer-core/src/main/java/io/micrometer/core/instrument/binder/mongodb/MongoMetricsConnectionPoolListener.java [hunk 1]
```diff
@@ -50,19 +50,25 @@
...
+
+    public MongoMetricsConnectionPoolListener(MeterRegistry registry, MongoMetricsConnectionPoolTagsProvider tagsProvider) {
         this.registry = registry;
+        this.tagsProvider = tagsProvider;
     }
 
     @Override
     public void connectionPoolOpened(ConnectionPoolOpenedEvent event) {
         List<Meter> connectionMeters = new ArrayList<>();
-        connectionMeters.add(registerGauge(event.getServerId(), METRIC_PREFIX + "size",
```
[the remark this answers is not in the document]
Comment (gustavomonarin) [reply micrometer-core/src/main/java/io/micrometer/core/instrument/binder/mongodb/MongoMetricsConnectionPoolListener.java:61]: sounds good for me, maybe as an improvement as part of the possible mongo 4 migration which would have the two current PR merged and follow the same strategy at once?

[the remark this answers is not in the document]
Comment (bono007) [reply micrometer-core/src/main/java/io/micrometer/core/instrument/binder/mongodb/MongoMetricsConnectionPoolListener.java:61]: Sure - it is a minor comment and not pressing.

[the remark this answers is not in the document]
Comment (bono007) [reply micrometer-core/src/main/java/io/micrometer/core/instrument/binder/mongodb/MongoMetricsConnectionPoolListener.java:61]: Sure - it is a minor comment and not pressing.

**Review by jonatan-ivanov — APPROVED** [review]

**Comment by jonatan-ivanov** [comment]
@gustavomonarin I modified the PR, merged master into it and migrated to use mongo 4.x.

**Comment by bono007** [comment]
Thank you @jonatan-ivanov for pushing this triplicate of PRs through to master!

**Pull request merged**